src/DotNetWorker.Grpc/GrpcWorker.cs

```diff
@@ -210,22 +211,18 @@ internal static WorkerInitResponse WorkerInitRequestHandler(WorkerInitRequest re
 
             response.WorkerMetadata.CustomProperties.Add("Worker.Grpc.Version", typeof(GrpcWorker).Assembly.GetName().Version?.ToString());
 
+            // Add required capabilities; these cannot be modified
             response.Capabilities.Add("RpcHttpBodyOnly", bool.TrueString);
             response.Capabilities.Add("RawHttpBodyBytes", bool.TrueString);
             response.Capabilities.Add("RpcHttpTriggerMetadataRemoved", bool.TrueString);
             response.Capabilities.Add("UseNullableValueDictionaryForHttp", bool.TrueString);
             response.Capabilities.Add("TypedDataCollection", bool.TrueString);
             response.Capabilities.Add("WorkerStatus", bool.TrueString);
-            response.Capabilities.Add("HandlesWorkerTerminateMessage", bool.TrueString);
-            response.Capabilities.Add("HandlesInvocationCancelMessage", bool.TrueString);
 
-            if (workerOptions.EnableUserCodeException)
+            // Add additional capabilites defined by WorkerOptions
+            foreach (KeyValuePair<string, string> entry in workerOptions.Capabilities)
```

Comment:
I think you can deconstruct KVP:

```suggestion
            foreach ((string key, string value) in workerOptions.Capabilities)
```

It might not be available in `netstandard2.0` though. But if you are interested, you can include it yourself. I tend to have a lang-compat file for netstandard2.0 projects: https://github.com/microsoft/durabletask-dotnet/blob/main/src/Shared/Core/CSharpLangCompat.cs

``` CSharp
// These extension methods are defined in the global namespace so that they're available everywhere.
internal static class KeyValuePairExtensions
{
    // Based on https://source.dot.net/#System.Private.CoreLib/KeyValuePair.cs,aa57b8e336bf7f59
    [EditorBrowsable(EditorBrowsableState.Never)]
    public static void Deconstruct<TKey, TValue>(this KeyValuePair<TKey, TValue> pair, out TKey key, out TValue value)
    {
        key = pair.Key;
        value = pair.Value;
    }
}
```

This is all if you are interested in that change. No worries if not interested.